internal/handler/web/people.go

```diff
@@ -33,6 +35,38 @@ func (h *Handler) GetPeople(ctx *gin.Context) {
 
 }
 
+func (h *Handler) PostPeopleId(ctx *gin.Context, id string) {
+	var update openapi.PersonUpdate
+	if err := ctx.ShouldBindJSON(&update); err != nil {
+		setInvalid(ctx, err, "Invalid JSON object")
+		return
+	}
+
+	// take the id from the url and get the person
+	person, err := h.users.GetPerson(id)
+	if err != nil {
+		setServerError(ctx, err, "Failed to get person")
+		return
+	}
+
+	switch update.TrainingKind {
+	case openapi.TrainingKindNhsd:
+		if err := h.users.SetNhsdTrainingValidity(person, update.TrainingDate); err != nil {
+			setServerError(ctx, err, "Failed to update training validity")
+		}
+
+	default:
+		panic("unsupported training kind")
+
+	}
+
+	ctx.JSON(http.StatusOK, openapi.TrainingRecord{
+		Kind:        update.TrainingKind,
+		CompletedAt: &update.TrainingDate,
+		IsValid:     true,
```

Comment:
the validity should be determined by the `TrainingDate` (service probably needs a new method for that). happy to do it in a followup